### ISSUE TITLE
Question: How to build googleapis stubs with go/go-grpc plugin?

### ISSUE THREAD

**Issue by adev73** [bufbuild/buf#427]
After following the tour guide to buf cli, I tried to build the googleapis using the Go configuration from the documentation.

My buf.yaml file is as follows:
```
version: v1beta1
build:
  roots:
    - .
lint:
  use:
    - BASIC
    - FILE_LOWER_SNAKE_CASE
  except:
    - ENUM_NO_ALLOW_ALIAS
    - IMPORT_NO_PUBLIC
    - PACKAGE_AFFINITY
    - PACKAGE_DIRECTORY_MATCH
    - PACKAGE_SAME_DIRECTORY
  ignore_only:
    ENUM_VALUE_UPPER_SNAKE_CASE:
      - google/cloud/sql/v1beta4/cloud_sql_resources.proto
    FIELD_LOWER_SNAKE_CASE:
      - google/cloud/sql/v1beta4/cloud_sql_tiers.proto
    ONEOF_LOWER_SNAKE_CASE:
      - google/actions/sdk/v2/event_logs.proto
      - google/cloud/aiplatform/v1beta1/explanation.proto
      - google/cloud/asset/v1/asset_service.proto
      - google/cloud/audit/audit_log.proto
      - google/cloud/security/privateca/v1/resources.proto
      - google/cloud/security/privateca/v1beta1/resources.proto
      - google/dataflow/v1beta3/jobs.proto
      - google/maps/fleetengine/v1/fleetengine.proto
breaking:
  use:
    - WIRE_JSON
```

This is my buf.gen.yaml file:
```
version: v1beta1
plugins:
  - name: cpp
    out: gen/cpp
  - name: go
    out: gen/go
    opt: paths=source_relative
  - name: go-grpc
    out: gen/go
    opt: paths=source_relative
```

### Environment 
- Linux Mint - 20.1, 
- buf - 0.51.1
- protoc - 3.6.1
- protoc-gen-go - v1.26.0
- protoc-gen-go_grpc  - 1.1.0

### Expected output:
- Stub files in C++ in gen/cpp
- Stub files in Go in gen/go

### Actual output:
- Thousands of warnings when generating cpp files, e.g.
```
google/cloud/speech/v1/cloud_speech.proto: warning: Import google/protobuf/any.proto but not used.
google/cloud/speech/v1/cloud_speech.proto: warning: Import google/protobuf/wrappers.proto but not used.
google/cloud/tasks/v2beta2/old_target.proto: warning: Import google/cloud/tasks/v2beta2/target.proto but not used.
```
However, the cpp stubs have been created & appear to be OK.

- Errors when generating go code & no stubs generated:
```
protoc-gen-go: unable to determine Go import path for "google/example/endpointsapis/v1/workspace.proto"

Please specify either:
	• a "go_package" option in the .proto source file, or
	• a "M" argument on the command line.

See https://developers.google.com/protocol-buffers/docs/reference/go-generated#package for more information.

protoc-gen-go: unable to determine Go import path for "google/cloud/common_resources.proto"

Please specify either:
	• a "go_package" option in the .proto source file, or
	• a "M" argument on the command line.

See https://developers.google.com/protocol-buffers/docs/reference/go-generated#package for more information.

Failure: plugin go: exit status 1; exit status 1.
```

This seems to be correct behaviour, the two files mentioned do not have an `options go_package = xxx` line. The second one (common_resources.proto) calls out in the comments that it's not meant to be generated, just used by other tools:
```
// This file contains stub messages for common resources in GCP.
// It is not intended to be directly generated, and is instead used by
// other tooling to be able to match common resource patterns.
```
### Questions
- Can I exclude certain files from the generate process, and if so & I do exclude them, will buf correctly generate the go stub files? 
- Can I specify options on a per-file basis, e.g. supplying the -M option to the go plugin call (although it seems that might be the wrong thing to do)?

I'm less bothered by the cpp warnings, as I don't use (and never intend to use) C++...

**Comment by johanbrandhorst** [comment]
Hi! Yeah unfortunately this is an issue with `googleapis` in general - it doesn't generate with the default `go` or `go-grpc` plugins. This is independent of `buf` and would happen with `protoc` too. Fortunately, most of the time users don't need to generate all the stubs for all the `googleapis`, so let me ask you, what are you trying to do?

If you only want to generate a few of the files from `googleapis`, you can specify paths and directories you're interested in with `--path` to `buf generate`.

Please let me know if you have anymore questions.

**Comment by adev73** [comment]
Hi,

Thanks for your answer! In truth, I was just experimenting, that's all :) I don't (yet!) have any need for the googleapi stubs, I was just curious to see if buf would build them in Go for me. I've only used protobuf at a very simplistic level, with no complex APIs, and certainly no need for cross referenced files, or imports, common libs etc. I guess if Google don't care that the plugin they wrote won't compile the apis they designed, then I probably shouldn't care either! 

Happy to close this Q, but I figured I'd ask it as I'm sure I'm not the first - nor will be the last - person to try this out...

Cheers,
Ade.
